test/basic.test.ts

```diff
@@ -2287,41 +2289,60 @@ describe('component islands', () => {
 
     if (isDev()) {
       const fixtureDir = normalize(fileURLToPath(new URL('./fixtures/basic', import.meta.url)))
-      for (const link of result.head.link) {
-        link.href = link.href!.replace(fixtureDir, '/<rootDir>').replaceAll('//', '/')
-        link.key = link.key!.replace(/-[a-z0-9]+$/i, '')
+      for (const head of result.head) {
+        for (const key in head) {
+          if (key === 'link') {
+            head[key] = head[key]?.map((h) => {
+              if (h.href) {
+                h.href = resolveUnrefHeadInput(h.href).replace(fixtureDir, '/<rootDir>').replaceAll('//', '/')
+              }
+              return h
+            })
+          }
+        }
       }
-      result.head.link.sort((a, b) => b.href!.localeCompare(a.href!))
     }
 
     // TODO: fix rendering of styles in webpack
     if (!isDev() && !isWebpack) {
       expect(normaliseIslandResult(result).head).toMatchInlineSnapshot(`
-        {
-          "link": [],
-          "style": [
-            {
-              "innerHTML": "pre[data-v-xxxxx]{color:blue}",
-              "key": "island-style",
-            },
-          ],
-        }
+        [
+          {
+            "style": [
+              {
+                "innerHTML": "pre[data-v-xxxxx]{color:blue}",
+              },
+            ],
+          },
+          {
+            "style": [
+              {
+                "innerHTML": "pre[data-v-xxxxx]{color:blue}",
+              },
+            ],
+          },
```

Comment:
need to investigate this

Reply:
That's because we don't use keys anymore